__tests__/app.test.js

```diff
@@ -12,22 +13,61 @@ beforeEach(() => {
-
-        })
-    })
-})
\ No newline at end of file
+describe("far-han-news tests", () => {
+  describe("/api/topics", () => {
+    test("GET 200: should return with an array of topics", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body }) => {
+          const { topics } = body;
+          expect(topics).toEqual(expect.any(Array));
+          expect(topics.length).toBe(3);
+          topics.forEach((topic) => {
+            expect(topic).toHaveProperty("slug", expect.any(String));
+            expect(topic).toHaveProperty("description", expect.any(String));
+          });
+        });
+    });
+    test("GET 404: should return an error for a path that does not exist", () => {
+      return request(app).get("/api/teepics").expect(404);
+    });
+  });
+  describe("/api/artiles/:article_id", () => {
+    test("GET 200: should get a valid article by its ID", () => {
+      return request(app)
+        .get("/api/articles/2")
+        .expect(200)
+        .then(({ body }) => {
+          const { article } = body;
+          expect(article[0]).toMatchObject({
+            article_id: 2,
+            title: "Sony Vaio; or, The Laptop",
+            topic: "mitch",
+            author: "icellusedkars",
+            body: "Call me Mitchell. Some years ago—never mind how long precisely—having little or no money in my purse, and nothing particular to interest me on shore, I thought I would buy a laptop about a little and see the codey part of the world. It is a way I have of driving off the spleen and regulating the circulation. Whenever I find myself growing grim about the mouth; whenever it is a damp, drizzly November in my soul; whenever I find myself involuntarily pausing before coffin warehouses, and bringing up the rear of every funeral I meet; and especially whenever my hypos get such an upper hand of me, that it requires a strong moral principle to prevent me from deliberately stepping into the street, and methodically knocking people’s hats off—then, I account it high time to get to coding as soon as I can. This is my substitute for pistol and ball. With a philosophical flourish Cato throws himself upon his sword; I quietly take to the laptop. There is nothing surprising in this. If they but knew it, almost all men in their degree, some time or other, cherish very nearly the same feelings towards the the Vaio with me.",
+            created_at: "2020-10-16T05:03:00.000Z",
+            votes: 0,
+            article_img_url:
+              "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+          });
+        });
+    });
+    test("GET 400: should return an error when passed an non existent ID of the correct type", () => {
+      return request(app)
+        .get("/api/articles/9999")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Bad request");
```

Comment:
This should be a 404. article id 9999 doesn't exist.... but it could do at some point! A 400 is a bad request. For example if a wrong data type is inputted